pkg/middleware/middleware.go

```diff
@@ -72,6 +74,28 @@ func Process(ctx *quaycontext.QuayRegistryContext, quay *v1.QuayRegistry, obj cl
 		return dep, nil
 	}
 
+	// If the current object is a PVC, check for volume override
+	if pvc, ok := obj.(*corev1.PersistentVolumeClaim); ok {
+		var volumeSizeOverride resource.Quantity
+		switch quayComponentLabel {
+		case "postgres":
+			volumeSizeOverride, err = getVolumeSizeOverrideForComponent(quay, v1.ComponentPostgres)
+			if err != nil {
+				return nil, fmt.Errorf("could not override volume size: " + err.Error())
+			}
+		case "clair-postgres":
+			volumeSizeOverride, err = getVolumeSizeOverrideForComponent(quay, v1.ComponentClair)
+			if err != nil {
+				return nil, fmt.Errorf("could not override volume size: " + err.Error())
+			}
+		}
+		if volumeSizeOverride.Cmp(*pvc.Spec.Resources.Requests.Storage()) == -1 {
```

Comment:
We shouldn't blindly trust that we can dereference the return of `Storage()`. Can you check if the returned value is not nil before calling `Cmp()` ? In this scenario I guess it is OK but it does not hurt to make this more "defensive".